PID.java

```diff
@@ -0,0 +1,32 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+public class PID extends Controller {
+
+    final private Collection<Controller> controllers;
+
+    private double wantedValue = 0;
+
+    public PID(final Controller... controllers) {
+        super(1);
+        this.controllers = Arrays.stream(controllers).collect(Collectors.toList());
+    }
+
+    public PID(final double k, final Controller... controllers) {
+        super(k);
+        this.controllers = Arrays.stream(controllers).collect(Collectors.toList());
+    }
+
+    public void setWanted(final double wantedValue) {
+        this.wantedValue = wantedValue;
+    }
+
+    @Override
+    public double getOutput(final double currentValue) {
+        final double error = wantedValue - currentValue;
+
+        return k * controllers.stream().mapToDouble(c -> c.getOutput(error)).reduce(0,
+                (acc, curr) -> acc + curr);
```

Comment:
I'm all for lambdas, but Java lambdas are rather awkward, and you already have the functions you need:
```suggestion
        return k * controllers.stream().map(Controller::getOutput).reduce(Double::sum);
```

Edit: Actually you can't do this, because you need to pass `error` to `getOutput`. Too bad we don't have currying, so you do need to create a lambda for this (or use loops).

If you're interested, [read about Java lambdas](https://blogs.oracle.com/javamagazine/behind-the-scenes-how-do-lambda-expressions-really-work-in-java) :slightly_smiling_face: